src/main/scala/com/guidewire/cda/TableReader.scala

```diff
@@ -224,27 +272,97 @@ class TableReader(clientConfig: ClientConfig) {
         val fullReduceTime = tableStopwatch.getTime - startReduceTime
         log.info(s"Reduce DataFrames for table '$tableName' for fingerprint '$schemaFingerprint', took ${(fullReduceTime / 1000.0).toString} seconds")
 
-        // Write each table and its schema to a CSV file to the configured location
-        val startWriteTime = tableStopwatch.getTime
-        val manifestTimestampForTable = manifestMap(tableName).lastSuccessfulWriteTimestamp
-        val tableDataFrameWrapperForMicroBatch = DataFrameWrapperForMicroBatch(tableName, schemaFingerprint,
-          manifestTimestampForTable, dataFrameForTable)
-        outputWriter.write(tableDataFrameWrapperForMicroBatch)
-        val fullWriteTime = tableStopwatch.getTime - startWriteTime
-        log.info(s"Wrote file(s) for table '$tableName' for fingerprint '$schemaFingerprint', took ${(fullWriteTime / 1000.0).toString} seconds")
+        // Since all output types (CSV, Parquet, JdbcRaw, JdbcMerged) share a common savepoints.json,
+        // make sure there are no schema change issues for JdbcRaw or JdbcMerged before writing data for
+        // this fingerprint to any of the target types.
+        val jdbcRawIsOk = if (clientConfig.outputSettings.saveIntoJdbcRaw) {
+          if (outputWriter.schemasAreConsistent(dataFrameForTable, clientConfig.jdbcConnectionRaw.jdbcSchema, tableName, schemaFingerprint,
+            clientConfig.jdbcConnectionRaw.jdbcUrl, clientConfig.jdbcConnectionRaw.jdbcUsername, clientConfig.jdbcConnectionRaw.jdbcPassword, spark, outputWriter.JdbcWriteType.Raw)) {
+            true
+          } else {
+            false
+          }
+        } else {
+          true
```

Comment:
IIUC, this is saying that if `clientConfig.outputSettings.saveIntoJdbcRaw` comes out to false, you still force-evaluate it to true any way? That doesn't make sense. Am I missing something?

Reply:
Your evaluation is correct. Maybe a variable name change would help? I see your point. The variables basically represent "nothing is broken; continue."

These variables (jdbcRawIsOk, jdbcMergedIsOk) are assumed true. They are only evaluated to false in the event there are schema differences between the file and database table that we can't handle. This is regardless of whether the config options are true or false.
The false return for the variables is our mechanism for bypassing all work since we can't support the schema differences and we can't correct the problem. 
We can set up a meeting to discuss if necessary.